Issue re-loading passive scan rules on add-on upgrade

### Describe the bug

<details>
<summary>Relevant stack trace from zap.log</summary>

    2024-06-26 12:51:30,663 [ZAP-cfu] INFO  ExtensionAutoUpdate - There is/are 2 newer addons
    2024-06-26 12:51:36,464 [ZAP-DownloadInstaller] INFO  ExtensionAutoUpdate - Installing new addon webdriverlinux v93.0.0
    2024-06-26 12:51:36,886 [ZAP-DownloadInstaller] INFO  ExtensionAutoUpdate - Finished installing new addon webdriverlinux v93.0.0
    2024-06-26 12:51:37,094 [ZAP-DownloadInstaller] INFO  ExtensionAutoUpdate - Installing new addon selenium v15.26.0
    2024-06-26 12:51:37,569 [ZAP-DownloadInstaller] WARN  PassiveScannerList - A scan rule with the name "Authentication Request Identified" already exists. The rule "org.zaproxy.addon.authhelper.AuthenticationDetectionScanRule" will not be loaded.
    2024-06-26 12:51:37,570 [ZAP-DownloadInstaller] ERROR ExtensionPassiveScan - Failed to install pscanrule: org.zaproxy.addon.authhelper.AuthenticationDetectionScanRule
    2024-06-26 12:51:37,570 [ZAP-DownloadInstaller] WARN  PassiveScannerList - A scan rule with the name "Session Management Response Identified" already exists. The rule "org.zaproxy.addon.authhelper.SessionDetectionScanRule" will not be loaded.
    2024-06-26 12:51:37,570 [ZAP-DownloadInstaller] ERROR ExtensionPassiveScan - Failed to install pscanrule: org.zaproxy.addon.authhelper.SessionDetectionScanRule
    2024-06-26 12:51:37,570 [ZAP-DownloadInstaller] WARN  PassiveScannerList - A scan rule with the name "Verification Request Identified" already exists. The rule "org.zaproxy.addon.authhelper.VerificationDetectionScanRule" will not be loaded.
    2024-06-26 12:51:37,570 [ZAP-DownloadInstaller] ERROR ExtensionPassiveScan - Failed to install pscanrule: org.zaproxy.addon.authhelper.VerificationDetectionScanRule
    2024-06-26 12:51:37,648 [ZAP-DownloadInstaller] INFO  ExtensionAutoUpdate - Finished installing new addon selenium v15.26.0
    2024-06-26 12:52:39,570 [AWT-EventQueue-0] ERROR UncaughtExceptionLogger - Exception in thread "AWT-EventQueue-0"
    java.lang.NoClassDefFoundError: org/zaproxy/addon/authhelper/AuthenticationRequestDetails
    	at org.zaproxy.addon.authhelper.AuthenticationDetectionScanRule.getExampleAlerts(AuthenticationDetectionScanRule.java:207) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.internal.ScanRulesInfo.addAlertRefs(ScanRulesInfo.java:103) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.internal.ScanRulesInfo.addEntry(ScanRulesInfo.java:75) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.internal.ScanRulesInfo.<init>(ScanRulesInfo.java:61) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.ExtensionAlertFilters.getScanRulesInfo(ExtensionAlertFilters.java:145) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.internal.ui.AlertSelectionPanel.<init>(AlertSelectionPanel.java:38) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.DialogAddAlertFilter.getAlertSelectionPanel(DialogAddAlertFilter.java:541) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.DialogAddAlertFilter.getFieldsPanel(DialogAddAlertFilter.java:310) ~[?:?]
    	at org.zaproxy.zap.view.AbstractFormDialog.initView(AbstractFormDialog.java:131) ~[zap-2.15.0.jar:2.15.0]
    	at org.zaproxy.zap.view.AbstractFormDialog.initialise(AbstractFormDialog.java:100) ~[zap-2.15.0.jar:2.15.0]
    	at org.zaproxy.zap.view.AbstractFormDialog.<init>(AbstractFormDialog.java:84) ~[zap-2.15.0.jar:2.15.0]
    	at org.zaproxy.zap.view.AbstractFormDialog.<init>(AbstractFormDialog.java:79) ~[zap-2.15.0.jar:2.15.0]
    	at org.zaproxy.zap.extension.alertFilters.DialogAddAlertFilter.<init>(DialogAddAlertFilter.java:109) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.DialogAddAlertFilter.<init>(DialogAddAlertFilter.java:99) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.ExtensionAlertFilters.showAddDialog(ExtensionAlertFilters.java:202) ~[?:?]
    	at org.zaproxy.zap.extension.alertFilters.ExtensionAlertFilters$1.performAction(ExtensionAlertFilters.java:178) ~[?:?]
    	at org.zaproxy.zap.extension.alert.PopupMenuItemAlert.performActions(PopupMenuItemAlert.java:120) ~[zap-2.15.0.jar:2.15.0]
    	at org.zaproxy.zap.extension.alert.PopupMenuItemAlert$PerformActionsActionListener.actionPerformed(PopupMenuItemAlert.java:222) ~[zap-2.15.0.jar:2.15.0]
    	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1972) ~[?:?]
    	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2313) ~[?:?]
    	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405) ~[?:?]
    	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262) ~[?:?]
    	at javax.swing.AbstractButton.doClick(AbstractButton.java:374) ~[?:?]
    	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:1028) ~[?:?]
    	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:1072) ~[?:?]
    	at java.awt.Component.processMouseEvent(Component.java:6626) ~[?:?]
    	at javax.swing.JComponent.processMouseEvent(JComponent.java:3389) ~[?:?]
    	at java.awt.Component.processEvent(Component.java:6391) ~[?:?]
    	at java.awt.Container.processEvent(Container.java:2266) ~[?:?]
    	at java.awt.Component.dispatchEventImpl(Component.java:5001) ~[?:?]
    	at java.awt.Container.dispatchEventImpl(Container.java:2324) ~[?:?]
    	at java.awt.Component.dispatchEvent(Component.java:4833) ~[?:?]
    	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4948) ~[?:?]
    	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4575) ~[?:?]
    	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4516) ~[?:?]
    	at java.awt.Container.dispatchEventImpl(Container.java:2310) ~[?:?]
    	at java.awt.Window.dispatchEventImpl(Window.java:2780) ~[?:?]
    	at java.awt.Component.dispatchEvent(Component.java:4833) ~[?:?]
    	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:775) ~[?:?]
    	at java.awt.EventQueue$4.run(EventQueue.java:720) ~[?:?]
    	at java.awt.EventQueue$4.run(EventQueue.java:714) ~[?:?]
    	at java.security.AccessController.doPrivileged(AccessController.java:399) ~[?:?]
    	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) ~[?:?]
    	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97) ~[?:?]
    	at java.awt.EventQueue$5.run(EventQueue.java:747) ~[?:?]
    	at java.awt.EventQueue$5.run(EventQueue.java:745) ~[?:?]
    	at java.security.AccessController.doPrivileged(AccessController.java:399) ~[?:?]
    	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) ~[?:?]
    	at java.awt.EventQueue.dispatchEvent(EventQueue.java:744) ~[?:?]
    	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) [?:?]
    	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) [?:?]
    	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) [?:?]
    	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) [?:?]
    	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:?]
    	at java.awt.EventDispatchThread.run(EventDispatchThread.java:90) [?:?]
    Caused by: java.lang.ClassNotFoundException
    	at org.zaproxy.zap.control.AddOnClassLoader.findClass(AddOnClassLoader.java:330) ~[zap-2.15.0.jar:2.15.0]
    	at java.lang.ClassLoader.loadClass(ClassLoader.java:592) ~[?:?]
    	at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
    	... 55 more

</details>

### Steps to reproduce the behavior

This occurred with ZAP on Kali at startup there were updates to install. I installed them, then when trying to create an Alert Filter the dialog would not display.

This was discussed on slack and the Alert Filter issue is secondary to the scan rule re-loading issue.

### Expected behavior

The scan rules should successfully unload/reload.

### Software versions

ZAP 2.15
Alert Filters 0.13.0

### Screenshots

N/A

### Errors from the zap.log file

See above.

### Additional context

N/A

### Would you like to help fix this issue?

- [ ] Yes

--- a/zap/src/main/java/org/zaproxy/zap/extension/pscan/ExtensionPassiveScan.java
+++ b/zap/src/main/java/org/zaproxy/zap/extension/pscan/ExtensionPassiveScan.java
@@ -709,6 +709,7 @@ public class ExtensionPassiveScan extends ExtensionAdaptor implements SessionCha
                     loadScanRules(statusUpdate.getAddOn());
                     break;
 
+                case SOFT_UNINSTALL:
                 case UNINSTALL:
                     AddOn addOn = statusUpdate.getAddOn();
 

